OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -781,6 +781,35 @@ public async Task<SearchResult<WorkshopDto>> FetchByFilterForAdmins(WorkshopFilt
         };
     }
 
+    /// <inheritdoc/>
+    public async Task<PriceRange> GetPriceRange(WorkshopFilter filter = null)
+    {
+        logger.LogDebug("Getting Price Range for workshops by filter started.");
+
+        filter ??= new WorkshopFilter();
+
+        var filterPredicate = PredicateBuild(filter);
+
+        var query = await workshopRepository.GetByFilter(
```

Comment:
`GetByFilter` return an enumerable, so technically this request will return 1000s workshops for you to process and receive OOM exception :). You can use `Get` to return queryable with the given `where` predicate, then do some SQL/Linq magic, e.g.:
```csharp
queriable
    .GroupBy(_ => 1)
    .Select(g => new PriceRange
    {
        MinPrice = g.Min(p => p.Price),
        MaxPrice = g.Max(p => p.Price)
    })
```

Though need to investigate if `GroupBy(_ => 1)` is still a thing to make 1 single query, cause it was a solution from [2018](https://github.com/dotnet/efcore/issues/13515)